CUJExecutor/app/src/androidTest/java/com/example/firstapp/ShellUtility.java

```diff
@@ -582,4 +582,24 @@ public void executeCUJ(String preStr, String postStr, int iterations, boolean re
             logData(allActionStamps);
         }
     }
+
+
+    public void singleCujRun(String preStr, String cujStr, boolean includeMeasured) throws Exception {
+        String[] preCUJ = parseToArray(preStr);
+        String[] postCUJ = parseToArray(cujStr);
+        String[] cujStrings;
+        if (includeMeasured) {
+            cujStrings = new String[preCUJ.length + postCUJ.length - 1];
+            System.arraycopy(preCUJ, 0, cujStrings, 0, preCUJ.length);
+            System.arraycopy(postCUJ, 0, cujStrings, preCUJ.length, postCUJ.length - 1);
+        } else {
+            cujStrings = preCUJ;
+        }
+
+        Action[] cuj = parseStringCUJ(cujStrings);
+
+        //caching run
+        cacheCUJ(cuj);
```

Comment:
Why use `cacheCUJ` to run the CUJ?  What does caching have to do with `singleCujRun`?

Reply:
just an easy way (given the implementation) to go through the CUJ 1 time. (obviously can't used cached data at all if we're only doing one run)

Reply:
If that is the case, then needs a comment.